### PR TITLE
yazi-unwrapped: use system jemalloc

### PR DESCRIPTION
**Pull request by hbenazha**
## Description of changes

Fixes the crash issue with Yazi on NixOS running on Apple Silicon.

Currently, Yazi uses a vendored jemalloc that's configured to work with 4K page size. However, since the Asahi Linux kernel on Apple Silicon uses 16K pages, Yazi crashes on startup with the error:
`<jemalloc>: Unsupported system page size`

I've taken the same approach as in issue #300025, where they use the Nixpkgs version of jemalloc that supports Apple Silicon. This solves the crash.

Maintainers ping: @XYenon @matthiasbeyer 
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ya/yazi-unwrapped/package.nix [hunk 1]
```diff
@@ -1,14 +1,13 @@
...
+  fetchFromGitHub,
+  lib,
+  installShellFiles,
+  stdenv,
+  Foundation,
+  rust-jemalloc-sys,
+  nix-update-script,
 }:
-
 rustPlatform.buildRustPackage rec {
```
Comment (matthiasbeyer) [inline pkgs/by-name/ya/yazi-unwrapped/package.nix:11]: Most of the diff here has nothing to do with the actual change in the commit.

File: pkgs/by-name/ya/yazi-unwrapped/package.nix [hunk 2]
```diff
@@ -37,13 +38,13 @@ rustPlatform.buildRustPackage rec {
...
 
-  passthru.updateScript = nix-update-script { };
+  passthru.updateScript = nix-update-script {};
 
   meta = {
     description = "Blazing fast terminal file manager written in Rust, based on async I/O";
     homepage = "https://github.com/sxyazi/yazi";
     license = lib.licenses.mit;
-    maintainers = with lib.maintainers; [ xyenon matthiasbeyer ];
+    maintainers = with lib.maintainers; [xyenon matthiasbeyer];
```
Comment (matthiasbeyer) [inline pkgs/by-name/ya/yazi-unwrapped/package.nix:47]: This has nothing to do with the actual change.

**Review by matthiasbeyer — CHANGES_REQUESTED** [review]
There's a lot of diff here for very little that's actually changed.

Please revert the changes that have nothing to do with your fix.

**Comment by hbenazha** [comment]
> There's a lot of diff here for very little that's actually changed.
> 
> Please revert the changes that have nothing to do with your fix.

Yes indeed, sorry, my formatter (Alejandra) formatted the whole file on save. I'm cleaning that.

**Review by matthiasbeyer — APPROVED** [review]

File: pkgs/by-name/ya/yazi-unwrapped/package.nix [hunk 3]
```diff
@@ -25,7 +26,8 @@ rustPlatform.buildRustPackage rec {
   env.YAZI_GEN_COMPLETIONS = true;
 
   nativeBuildInputs = [ installShellFiles ];
-  buildInputs = lib.optionals stdenv.isDarwin [ Foundation ];
+  buildInputs = lib.optionals stdenv.isDarwin [Foundation]
+                ++ lib.optionals (!stdenv.isDarwin) [rust-jemalloc-sys];
```
Comment (XYenon) [inline pkgs/by-name/ya/yazi-unwrapped/package.nix:30]: It looks like `rust-jemalloc-sys` is also available on darwin, maybe we can make it always be used?

```suggestion
  buildInputs = [ rust-jemalloc-sys ] ++ lib.optionals stdenv.isDarwin [ Foundation ];
```

**Review by XYenon — CHANGES_REQUESTED** [review]

File: pkgs/by-name/ya/yazi-unwrapped/package.nix (hunk 3 above)
Comment (XYenon) [reply pkgs/by-name/ya/yazi-unwrapped/package.nix:30]: I have tried it on `x86_64-darwin`, and it works.

Comment (hbenazha) [reply pkgs/by-name/ya/yazi-unwrapped/package.nix:30]: Thank you for trying it on darwin. I'll commit your change

**Review by XYenon — APPROVED** [review]

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**